src/test/java/it/cnr/isti/hpc/wikipedia/article/en/ArticleTest.java

```diff
@@ -93,19 +95,116 @@ public void testNoEmptyAnchors() throws IOException {
 
         // No anchor should be empty
         for (Link link:a.getLinks()){
-            assertFalse(link.getDescription().isEmpty());
+            assertFalse(link.getAnchor().isEmpty());
         }
 
         // testing an specific anchor
         for (Link link:a.getLinks()){
             if (link.getId().equals("HTMS_Chakri_Naruebet"))
-                assertEquals(link.getDescription(),"HTMS Chakri Naruebet");
+                assertEquals(link.getAnchor(),"HTMS Chakri Naruebet");
         }
 
+    }
+    
+    @Test
+    public void testNoEmptyWikiIds() throws IOException {
+        Article a = new Article();
+        String mediawiki = IOUtils.getFileAsUTF8String("./src/test/resources/en/Cenozoic");
+        parser.parse(a, mediawiki);
 
+        for(Link l: a.getLinks()){
+            assert(!l.getId().equals(""));
+        }
+    }
+    
+    @Test
+    public void testEmptyLinksShouldBeFiltered() throws IOException {
+        // Some annotations are incomplete on wikipedia i.e: [[]] [[ ]]
+        // Those should be filtered
+        Article a = new Article();
+        String mediawiki = IOUtils.getFileAsUTF8String("./src/test/resources/en/Phantom_kangaroo");
+        parser.parse(a, mediawiki);
+
+        for(Link l: a.getLinks()){
+            assert(!l.getId().equals(""));
```

Comment:
also, java asserts fail only if you enable asserts with `java -ea` otherwise they will be super silent (...) I would use JUNIT ``assertTrue ``